crates/sui-json-rpc/src/event_api.rs

```diff
@@ -107,7 +107,18 @@ impl EventReadApiServer for EventReadApi {
         let state = self.state.clone();
         let stream = self.event_handler.subscribe(filter);
         let stream = stream.map(move |e: EventEnvelope| {
-            let event = SuiEvent::try_from(e.event, state.module_cache.as_ref());
+            let event = SuiEvent::try_from(
```

Comment:
In this cas, is the right thing to do to recreate the event handler on each epoch as well?

Reply:
yes, we should take care of that too.

Reply:
I talked to @patrickkuo about this and there is work around events that could make a lot of this change.
I also had a conversation with @amnn about this and it's a bit tricky if clients are making calls across epoch.
In that respect we may need a cache that goes from epoch to module cache per epoch, but we'll follow up with Patrick on that.

Reply:
actually I think swapping the cache in the event handler is the right thing to do here.
I'll open an issue for that...

Reply:
I actually removed the caching of the `module_cache` from the EventHandler and we are now always passing it in from authority* and this should be fixed.
@patrickkuo nothing to do here, we should be fine